### ISSUE TITLE
0.3.0 release

### ISSUE THREAD

**Issue by pc494** [pyxem/orix#58]
As mentioned https://github.com/pyxem/pyxem/pull/596 we have 'enough' for a new minor version of `orix`, this issue serves to collate developer request for things that it would be good to have in by the release point - @hakonanes 

From my perspective

- [x] Clean up the deprecations from the `0.2` series 
- [x] Improve the documentation for `OrientationRegion`
- [x] Address #57 
- [x] Address the review comment on #60
- [x] Add reader/writer for CrystalMap #59
- [x] check the "grain-exchange" (GE) case of the `.equivalent` method, add path for `S1==S2` no GE

I'll add more things as they occur to me.

**Comment by dnjohnstone** [comment]
@pc494 and @hakonanes - could we set a target date for 0.3.0 release, please?

I'm thinking end of June might be realistic?

Also - do we think we want to put #62 in this one?

**Comment by hakonanes** [comment]
Yes, #62 please. I would like to couple it with stereopgrahic projections of EBSD master patterns in kikuchipy's `EBSDMasterPattern` class. I want to use orix a lot more in general in kikuchipy v0.3 (currently just released v0.2), so doing #59 is high on my list. End of June sounds good to me.

**Comment by dnjohnstone** [comment]
Not necessarily for 0.3.0 - but for the future - I think we should think about which packages we currently know of wanting to use orix upstream and what for.

In particular we have - pyxem, kikuchipy and diffsims , which would all benefit from different parts of orix functionality.

Not sure I'm actually saying anything here - just trying to map things out in my head an thought it might be good to make some kind of a plan.

**Comment by hakonanes** [comment]
Desired uses for kikuchipy, from the top of my head:
* Import orientation data along an EBSD scan into a `CrystalMap` object available as an attribute `s.crystal_map` or just `s.cm`. Then, we can make all maps available as Signal2D navigators e.g., to inform analysis. Also, to create masks, e.g. get one averaged pattern per grain ID etc.
* Store crystal data in `Phase` objects via metadata.
* Save quality maps or other maps produced from diffraction patterns via a `CrystalMap` HDF5 writer, for general saving and use in e.g. MTEX. (When the HDF5 reader/writer is done, I plan to make a PR to MTEX with an orix reader. An orix writer would also be nice, but we'll see.)
* Orientation sampling (have to move stuff over from pyxem to orix for this? Quite a lot of work?), to get an orientation library to create a diffraction pattern library from an EBSD master pattern.
* Keep track of crystal and sample coordinate systems, and change between them.

In general, when starting to use orix more, I (we) should be aware of what should go in kikuchipy (and pyxem etc.), and what should go into orix. There might be stuff both packages want but don't fit into orix... I guess I'll have to consider kikuchipy depending on pyxem or some other solution.

**Comment by dnjohnstone** [comment]
>     * Orientation sampling (have to move stuff over from pyxem to orix for this? Quite a lot of work?), to get an orientation library to create a diffraction pattern library from an EBSD master pattern.

This is currently in diffsims for the pyxem pipeline because that's where we do all the simulations of diffraction data. But good news is that diffsims is quite lightweight in terms of dependencies so you could consider depending on that in kikuchipy (if you depend on pyxem you already will anyway)...

@pc494 and I have discussed moving that stuff over to orix and I think it will happen eventually but for now it's been relatively low priority because we have a complete pipeline.

But definitely good to discuss that one with Phillip before starting on it and he did quite a lot on that already!

**Comment by hakonanes** [comment]
Ah yes, forgot it's there instead, will look into that. (Have actually removed the pyxem dependency in v0.2.)

**Comment by dnjohnstone** [comment]
>     * Store crystal data in `Phase` objects via metadata.

This again might be considered in terms of what we already have in diffsims where we use diffpy.Structure objects to create a StructureLibrary that we then use to initialize our diffraction simulations.

It would be nice if that Phase object was the same at the start and the end of the pipeline - so I think in the existing pyxem case we'd want that to be a diffpy.Structure object containing all the crystal data.

**Comment by dnjohnstone** [comment]
I think what I'm really getting to here is that it would make a lot of sense if the start and the end of the pipeline in terms of both structural data and prior orientation lists were based on the same objects. This should also ideally be the same for pyxem and kikuchipy because it just is the same information.

This would avoid a whole load of potential bugs where we update one end but not the other and reduce the total code maintenance burden.

**Comment by hakonanes** [comment]
Yes. So far, the `Phase` object only contains a name, crystal symmetry, and color. You two (@dnjohnstone, @pc494) have the exprience with using diffpy, how would you the `Phase` object change? I think it should just be a place to store and keep track of the information in diffpy.structure objects, and change the information when some `CrystalMap` or other class method requires it (don't have a concrete example).

**Comment by pc494** [comment]
(conversation about doing gridding in orix)

The short answer is yes, we/I should move this from diffsims to orix. I think it shouldn't be quiet as awful as I had thought it would be either. Although there may be some speed issues. Ideally would delay that until a 0.4 release.

(conversation about storing phases)

I'm not so sure about this, adding diffpy makes some presumptions about how one is going to run their simulations and I'm not sure if we really need to do that.

**Comment by dnjohnstone** [comment]
Yeah, I agree with all of that - i.e. grid generation being beyond 0.3.0 and my suggestion about diffpy.Structure not being a great one on reflection...

**Comment by dnjohnstone** [comment]
Having thought about this some more, and slightly prompted by #62, I think we definitely need objects for storing crystal structures associated with phases. Because ultimately we might want to have analyses of orientation data that needs that.

I think that using the same object for that as we do elsewhere for simulations would make sense?

**Comment by pc494** [comment]
The thinking being that if you want to rotate to a [101] say you need to know `a,b,c,alpha,beta` and `gamma`?

**Comment by dnjohnstone** [comment]
For example, or I think if you want to make an IPF plot for any crystal?

Or anything in the crystal geometry bit of MTEX e.g. https://mtex-toolbox.github.io/FundamentalSector.html

**Comment by pc494** [comment]
Yep, I agree with you on this.

**Comment by hakonanes** [comment]
If you specify which API/calls you would prefer, I could have a go at using diffpy.structure (while looking at how you use it in pyxem/diffsims) in the `Phase` class, make a PR and see what you think? I can do it late today or tomorrow.

**Comment by hakonanes** [comment]
Or, since you are working on this now, @dnjohnstone, you could do it. Whatever you find best.

**Comment by dnjohnstone** [comment]
@pc494 did the grain exchange thing fall out in the wash or is there an action needed on that?

**Comment by pc494** [comment]
It still needs fixing, I'll do it and the other box in this issue and close in the next couple of days

**Comment by dnjohnstone** [comment]
Ok I think we're ready for orix-0.3.0 release.

I want to update the copyright statements because the dates are off. And then I'll release tomorrow unless anyone objects @hakonanes and @pc494 ?
